### PR TITLE
Create migrations and DataLayer projects. Add test migrations

### PR DESCRIPTION
**Pull request by VidasBui**

### CODE REVIEW

**Comment by radislavkireev** [comment]
Dup folder added.

File: src/CSI.IBTA.DB.Migrations/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace CSI.IBTA.DB.Migrations
+{
```
Comment (radislavkireev) [inline src/CSI.IBTA.DB.Migrations/Class1.cs]: could be removed

**Comment by radislavkireev** [comment]
Keep migrations in migrations assembly. Check sample https://github.com/dotnet/EntityFramework.Docs/blob/main/samples/core/Schemas/ThreeProjectMigrations/WebApplication1.sln

File: src/CSI.IBTA.DataLayer/Models/CsiHealthcare2024Context.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+
+namespace CSI.IBTA.DataLayer.Models;
+
+public partial class CsiHealthcare2024Context : DbContext
```
Comment (radislavkireev) [inline src/CSI.IBTA.DataLayer/Models/CsiHealthcare2024Context.cs]: Lets rename it to UserManagement context

File: src/CSI.IBTA.DataLayer/Models/CsiHealthcare2024Context.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    public CsiHealthcare2024Context(DbContextOptions<CsiHealthcare2024Context> options)
+        : base(options)
+    {
+    }
+
+    public virtual DbSet<Test> Tests { get; set; }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder.UseSqlServer("Server=(localdb)\\MSSQLLocalDB;Database=csi-healthcare-2024;Trusted_Connection=True;Encrypt=False;",
```
Comment (radislavkireev) [inline src/CSI.IBTA.DataLayer/Models/CsiHealthcare2024Context.cs]: Conn string should be extracted to the config

**Review by radislavkireev — CHANGES_REQUESTED** [review]

File: src/CSI.IBTA.DB.Migrations/Migrations/20240206210505_AddNumberToTestTable.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
```
Comment (radislavkireev) [inline src/CSI.IBTA.DB.Migrations/Migrations/20240206210505_AddNumberToTestTable.cs]: Delete testing migration

File: src/CSI.IBTA.DB.Migrations/Migrations/CsiHealthcare2024ContextModelSnapshot.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
+﻿// <auto-generated />
+using System;
+using CSI.IBTA.DataLayer.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+
+#nullable disable
+
```
Comment (radislavkireev) [inline src/CSI.IBTA.DB.Migrations/Migrations/CsiHealthcare2024ContextModelSnapshot.cs]: Delete testing context

**Review by radislavkireev — CHANGES_REQUESTED** [review]

**Review by radislavkireev — APPROVED** [review]

**Review by devJUKI — APPROVED** [review]

**Pull request merged**